src/layout/layout.js

```diff
@@ -191,21 +191,27 @@
    * @private
    */
   MaterialLayout.prototype.screenSizeHandler_ = function() {
-    if (this.screenSizeMediaQuery_.matches) {
-      this.element_.classList.add(this.CssClasses_.IS_SMALL_SCREEN);
-
-      if (this.drawer_) {
-        this.drawer_.setAttribute('aria-hidden', 'true');
-      }
+    // modified to query dependent elements rather than binding materialLayout to windows media query result
+    var materialLayout = window.document.querySelector('.mdl-layout')
+    if (materialLayout){
+      var drawerElement = materialLayout.querySelector('.mdl-layout__drawer');
+      if (window.matchMedia('(max-width: 1024px)').matches) {
```

Comment:
can't we check `.matches` on the change event?
https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/change_event